modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/ImageMosaicReaderTest.java

```diff
@@ -4632,6 +4634,40 @@ public void testCoverageOnBands() throws Exception {
         reader.dispose();
     }
 
+    @Test
+    public void testHeterogeneousConfigs() throws Exception {
+        // we have 2 mosaics here:
```

Comment:
Two mosaics... or two coverages in the same mosaic?

Reply:
Right, they are actually 2 coverages of the same mosaic store.